src/lib/Types/Core.hs

```diff
@@ -2156,3 +2157,6 @@ instance SubstE AtomSubstVal (EffectP AtomName) where
       RWSEffect rws v'
     ExceptionEffect -> ExceptionEffect
     IOEffect        -> IOEffect
+    UserEffect v    ->
+      let (Rename v') = env ! v
+       in UserEffect v'
```

Comment:
I think it would be better to use `case env ! v` here, since `let` allows the use of fallible patterns, whereas `case` would complain if you missed a constructor.